### PR TITLE
feat: add rand_u64_with_count and rand_u128_with_count

### PR DESCRIPTION
**Pull request by yubing744**
## Summary

Fix Calling simple_rng::seed() multiple times in the same transaction returns the same value

- Closes #issue
https://github.com/rooch-network/rooch/issues/2837

### CODE REVIEW

**Comment by yubing744** [comment]
@jolestar help review.

**Comment by yubing744** [comment]
@jolestar help review it again.

File: frameworks/rooch-framework/sources/simple_rng.move [hunk 1]
```diff
@@ -75,32 +75,70 @@ module rooch_framework::simple_rng {
...
 
+    fun seed_with_count(count: u64): vector<u8> {
+        let base_seed = seed();
+        let count_bytes = bcs::to_bytes(&count);
+        let i = 0;
+        while (i < 8) {
+            let value = (*vector::borrow(&base_seed, i)) ^ (*vector::borrow(&count_bytes, i));
+            vector::insert(&mut base_seed, i, value);
+            i = i + 1;
+        };
```
Comment (jolestar) [inline frameworks/rooch-framework/sources/simple_rng.move:86]: Use `vector::append`?

Comment (yubing744) [reply frameworks/rooch-framework/sources/simple_rng.move:86]: I tested and found that bytes_to_u64 only takes the first 8 bytes of the output bytes, so appending directly has no effect.

Comment (jolestar) [reply frameworks/rooch-framework/sources/simple_rng.move:86]: We should update the seed function and add the count to the data before hashing.

Comment (yubing744) [reply frameworks/rooch-framework/sources/simple_rng.move:86]: Directly modifying the seed function won’t cause compatibility issues?

Comment (jolestar) [reply frameworks/rooch-framework/sources/simple_rng.move:86]: The seed function is a private function; no compatibility issue

Comment (yubing744) [reply frameworks/rooch-framework/sources/simple_rng.move:86]: OK

Comment (yubing744) [reply frameworks/rooch-framework/sources/simple_rng.move]: @jolestar help review it again.

**Review by jolestar — APPROVED** [review]

**Pull request merged**